### PR TITLE
mitosheet: handle unset colors in conditional formatting

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

Fixes https://github.com/mito-ds/mito/issues/1274

# Testing

Please provide a list of the ways you can "access" or use the functionality. Please try and be exhaustive here, and make sure that you test everything you list.

- [ ] I have tested this on real data that is reasonable and large
- [ ] If I changed the interaction with JupyterLab, I tested that it does not break other programs (like VS Code), and tested that it works "multiple times" in the same notebook.

# Documentation

Note if any new documentation needs to addressed or reviewed.

### CODE REVIEW

File: mitosheet/src/mito/pro/taskpanes/SetDataframeFormat/SetDataframeFormatTaskpane.tsx [hunk 1]
```diff
@@ -80,6 +80,7 @@ const SetDataframeFormatTaskpane = (props: SetDataframeFormatTaskpaneProps): JSX
     }
 
     const updateDataframeFormatParams = (newParams: RecursivePartial<DataframeFormat>): void => {
+        console.log(newParams)
```
Comment (aarondr77) [inline mitosheet/src/mito/pro/taskpanes/SetDataframeFormat/SetDataframeFormatTaskpane.tsx:83]: remove

File: mitosheet/src/mito/pro/taskpanes/ConditionalFormatting/ConditionalFormattingTaskpane.tsx [hunk 2]
```diff
@@ -125,6 +125,19 @@ const ConditionalFormattingTaskpane = (props: ConditionalFormattingTaskpaneProps
...
+        newParams.conditional_formats = newParams?.conditional_formats?.map(conditionalFormat => {
+            const newConditionalFormat = {...conditionalFormat};
+            if (newConditionalFormat.color === "var(--mito-text)") {
+                newConditionalFormat.color = undefined;
+            }
+            if (newConditionalFormat.backgroundColor === "var(--mito-background-off)") {
+                newConditionalFormat.backgroundColor = undefined;
+            }
+            return newConditionalFormat;
+        })
```
Comment (naterush) [inline mitosheet/src/mito/pro/taskpanes/ConditionalFormatting/ConditionalFormattingTaskpane.tsx:139]: This is a weird way to do this. Why not just default them to undefined when we create them?

**Review by naterush — COMMENTED** [review]
I think this approach could be more clean and maintainable, unless I'm misunderstanding something.

**Comment by naterush** [comment]
This actually isn't a solution

**Pull request closed without merging**